### PR TITLE
Add rust bindings for zeroex smart contract

### PR DESCRIPTION
**Pull request by MartinquaXD**
In order to implement correct settlement encoding for a future `ZeroExInteraction` we need to have rust bindings for the zeroex smart contract.

Unfortunately the 0x team provides their [JSON ABI file](https://github.com/0xProject/protocol/blob/development/packages/contract-artifacts/artifacts/IZeroEx.json) in a format which is not immediately compatible with out `vendor.rs` script.

To support more formats I modified the logic to retrieve the required values a little.
Before we expected a flat JSON file where we only kept the properties `abi`, `devdoc`, `userdoc` and `bytecode`.
The 0x file is more deeply nested so we need a way to check multiple paths with different levels of nesting for a given target property.
Because the `vendor` script only gets minimal use I didn't want to waste time overengineering the solution and ended up with a syntax of `("abi", "abi,compilerOutput.abi")` which maps the first non-null value of `json.abi` or `json.compilerOutput.abi` to the property of `abi` in the resulting file.

### Test Plan
Check that we can generate bindings for the new file.

### CODE REVIEW

**Review by vkgnosis — APPROVED** [review]
LGTM. Should probably wait for Nick to review this too as he knows this type of thing best.

File: crates/contracts/src/bin/vendor.rs [hunk 1]
```diff
@@ -147,7 +151,12 @@ impl Vendor {
     fn full(&self) -> VendorContext {
         VendorContext {
             artifacts: &self.artifacts,
-            properties: &["abi", "bytecode", "devdoc", "userdoc"],
+            properties: &[
+                ("abi", "abi,compilerOutput.abi"),
+                ("devdoc", "devdoc,compilerOutput.devdoc"),
```
Comment (nlordell) [inline crates/contracts/src/bin/vendor.rs:156]: nice compromise.

**Review by nlordell — CHANGES_REQUESTED** [review]
Just requesting change on the vendoring source (i.e. can we use a tag or commit hash). Otherwise looks great.

File: crates/contracts/src/bin/vendor.rs [hunk 2]
```diff
@@ -125,6 +125,10 @@ fn run() -> Result<()> {
         .npm(
             "IUniswapV3Factory",
             "@uniswap/v3-core@1.0.0/artifacts/contracts/interfaces/IUniswapV3Factory.sol/IUniswapV3Factory.json",
+        )?
+        .github(
+            "IZeroEx",
+            "0xProject/protocol/development/packages/contract-artifacts/artifacts/IZeroEx.json",
```
[the remark this answers is not in the document]
Comment (MartinquaXD) [reply crates/contracts/src/bin/vendor.rs]: Good idea. Pinned the most recent commit of the file.

**Review by nlordell — APPROVED** [review]

**Pull request merged**